test/cpp/end2end/ssl_credentials_test.cc

```diff
@@ -85,51 +131,301 @@ class SslCredentialsTest : public ::testing::Test {
+    if (!result.ok()) {
+      return absl::Status(static_cast<absl::StatusCode>(result.error_code()),
+                          result.error_message());
+    }
+    EXPECT_EQ(response.message(), kMessage);
+    return context.auth_context();
+  }
+
+  void ExpectResumedSession(std::shared_ptr<const AuthContext> auth_context) {
+    std::vector<grpc::string_ref> properties =
+        auth_context->FindPropertyValues(GRPC_SSL_SESSION_REUSED_PROPERTY);
+    ASSERT_EQ(properties.size(), 1u);
+    EXPECT_EQ("true", ToString(properties[0]));
+  }
+
+  void ExpectNonResumedSession(
+      std::shared_ptr<const AuthContext> auth_context) {
+    std::vector<grpc::string_ref> properties =
+        auth_context->FindPropertyValues(GRPC_SSL_SESSION_REUSED_PROPERTY);
+    ASSERT_EQ(properties.size(), 1u);
+    EXPECT_EQ("false", ToString(properties[0]));
+  }
+
+  void ExpectOk(const absl::Status& status) {
+    EXPECT_EQ(status.code(), absl::StatusCode::kOk);
+    EXPECT_EQ(status.message(), "");
+  }
+
+  void ExpectTransportSecurityType(
+      std::shared_ptr<const AuthContext> auth_context) {
+    std::vector<grpc::string_ref> properties = auth_context->FindPropertyValues(
+        GRPC_TRANSPORT_SECURITY_TYPE_PROPERTY_NAME);
+    ASSERT_EQ(properties.size(), 1u);
+    EXPECT_EQ(ToString(properties[0]), GRPC_SSL_TRANSPORT_SECURITY_TYPE);
+  }
+
+  void ExpectTransportSecurityLevel(
+      std::shared_ptr<const AuthContext> auth_context) {
+    std::vector<grpc::string_ref> properties = auth_context->FindPropertyValues(
+        GRPC_TRANSPORT_SECURITY_LEVEL_PROPERTY_NAME);
+    ASSERT_EQ(properties.size(), 1u);
+    EXPECT_EQ(ToString(properties[0]), "TSI_PRIVACY_AND_INTEGRITY");
+  }
+
+  void ExpectPeerCommonName(std::shared_ptr<const AuthContext> auth_context,
+                            absl::string_view common_name) {
+    std::vector<grpc::string_ref> properties =
+        auth_context->FindPropertyValues(GRPC_X509_CN_PROPERTY_NAME);
+    ASSERT_EQ(properties.size(), 1u);
+    EXPECT_EQ(ToString(properties[0]), common_name);
+  }
+
+  void ExpectPeerSubject(std::shared_ptr<const AuthContext> auth_context,
+                         absl::string_view subject) {
+    std::vector<grpc::string_ref> properties =
+        auth_context->FindPropertyValues(GRPC_X509_SUBJECT_PROPERTY_NAME);
+    ASSERT_EQ(properties.size(), 1u);
+    EXPECT_EQ(ToString(properties[0]), subject);
+  }
+
+  void ExpectPeerSubjectAltName(std::shared_ptr<const AuthContext> auth_context,
+                                const absl::flat_hash_set<std::string>& sans) {
+    std::vector<grpc::string_ref> properties =
+        auth_context->FindPropertyValues(GRPC_X509_SAN_PROPERTY_NAME);
+    ASSERT_EQ(properties.size(), sans.size());
+    absl::flat_hash_set<std::string> observed_sans;
+    for (size_t i = 0; i < properties.size(); ++i) {
+      observed_sans.insert(ToString(properties[i]));
+    }
+    EXPECT_EQ(sans, observed_sans);
```

Comment:
nit: reverse the order to be consistent in this test file

Reply:
Done.